modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Traverse.kt

```diff
@@ -5,11 +5,526 @@ import arrow.core.Id
+ * }
+ * ```
+ *
+ * Interestingly enough, [traverse] is a much more generalized and powerful form of [sequence] that would allow us to parse a `List<String>` or validate credentials for a `List<User>`.
+ *
+ * Enter [Traverse].
+ *
+ * ### The Typeclass
+ *
+ * At center stage of Traverse is the [traverse] method.
+ *
+ * ```kotlin
+ * import arrow.Kind
+ * import arrow.typeclasses.Applicative
+ * import arrow.typeclasses.Foldable
+ * import arrow.typeclasses.Functor
+ *
+ * interface Traverse<F> : Functor<F>, Foldable<F> {
+ *   fun <G, A, B> Kind<F, A>.traverse(AP: Applicative<G>, f: (A) -> Kind<G, B>): Kind<G, Kind<F, B>>
+ * }
+ * ```
+ *
+ * In our above example, `F` is `List`, and `G` is `Option` or `IO`. For the profile example, traverse says given a `List<User>` and a function `(User) -> IO<Profile>`, it can give you a `IO<List<Profile>>`.
+ *
+ * Abstracting away the `G` (still imagining `F` to be `List`), [traverse] says given a collection of data, and a function that takes a piece of data and returns an value `B` wrapped in a container `G`, it will traverse the collection, applying the function and aggregating the values (in a `List`) as it goes.
+ *
+ * In the most general form, `Kind<F, A>` is some sort of context `F` which may contain a value `A` (or several). While `List` tends to be among the most general cases, there also exist [Traverse] instances for [Option], [Either], and [Validated] (among others).
+ *
+ * In general this holds:
+ *
+ * ```kotlin
+ * map(::f).sequence(AP) == traverse(AP, ::f)
+ * ```
+ *
+ * where AP stands for an `Applicative<G>`, which is in prior snippets `Applicative<ForOption>` or `Applicative<ForIO>`.
+ *
+ * ### Sequencing effects
+ *
+ * Sometimes our effectful functions return a [Unit] value in cases where there is no interesting value to return (e.g. writing to some sort of store).
+ *
+ * ```kotlin:ank
+ * import arrow.fx.IO
+ *
+ * interface Data
+ *
+ * fun writeToStore(data: Data): IO<Unit> = TODO("")
+ * ```
+ *
+ * If we traverse using this, we end up with a funny type.
+ *
+ * ```kotlin:ank
+ * import arrow.core.ListK
+ * import arrow.fx.IO
+ * import arrow.fx.extensions.io.applicative.applicative
+ * import arrow.fx.fix
+ *
+ * interface Data
+ *
+ * fun writeToStore(data: Data): IO<Unit> = TODO("")
+ * //sampleStart
+ * fun writeManyToStore(data: ListK<Data>): IO<ListK<Unit>> =
+ *   data.traverse(IO.applicative()) { writeToStore(it) }.fix()
+ * //sampleEnd
+ * ```
+ *
+ * We end up with a `IO<ListK<Unit>>`! A `ListK<Unit>` is not of any use to us, and communicates the same amount of information as a single [Unit] does.
+ *
+ * Traversing solely for the sake of the effects (ignoring any values that may be produced, [Unit] or otherwise) is common, so [Foldable] (superclass of [Traverse]) provides [traverse_] and [sequence_] methods that do the same thing as [traverse] and [sequence] but ignores any value produced along the way, returning [Unit] at the end.
```

Comment:
```suggestion
 * Traversing solely for the sake of the effects (ignoring any values that may be produced, [Unit] or otherwise) is common, so [Foldable] (superclass of [Traverse]) provides [traverse_] and [sequence_] methods that do the same thing as [traverse] and [sequence] but ignore any value produced along the way, returning [Unit] at the end.
```